app/services/who_report_service.rb

```diff
@@ -0,0 +1,127 @@
+class WhoReportService
+  attr_reader :region, :period, :months, :repo, :dashboard_analytics
+  def initialize(region, period)
+    @region = region
+    @period = period
+    @months = period.downto(5).reverse
+    regions = region.facility_regions.to_a << region
```

Comment:
I would suggest teaching FacilityDistrict how to return `facility_regions` to match so this will work...that will it will match the "Region-ish" interface.

```
class FacilityDistrict
  def facility_regions
    Region.facility_regions.where(source_id: facility_Ids)
  end
```

Reply:
Sounds good 👍